### DISCUSSION TITLE
MacOS Startup Options

### DISCUSSION THREAD

**Discussion by MarcioCantante** [tiny-pilot/tinypilot#534] [Q&A]
Hi!

Before I go further, thank you for making this gadget! TinyPilot is extremely helpful!

I was wondering if you've used the tool for MacOS startup key combinations, like Command (⌘)-R and Option (⌥) to start up in recovery mode or opening startup manager. I've added a couple Meta combination keys but it doesn't seem to work during startup.

**Comment by mtlynch** [comment]
Oh, I've never heard of users running into this. I unfortunately don't have a Mac on hand to test. Can you share your [debug logs](https://github.com/mtlynch/tinypilot#diagnostics)?

**Reply by MarcioCantante** [reply]
Sure! Here's the logs: https://logs.tinypilotkvm.com/wz1kFgEt.

Found a fix for the video loss using [uhubctl](https://github.com/mvp/uhubctl). 
Cutting the power then turning the ports on again seems to fix it.

**Reply by mtlynch** [reply]
Thanks for sharing the logs!

I'm afraid that it might be incompatible with MacOS startup keys. It looks like TinyPilot is recognizing the keys but the Mac is rejecting them.

>Found a fix for the video loss using uhubctl.
Cutting the power then turning the ports on again seems to fix it.

Oh, that's a good tip! Thanks for passing this along.
